uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php

```diff
@@ -734,6 +734,30 @@ public function upgrade_1032() {
     return TRUE;
   }
 
+  /**
+   * Update permissions and set the weight for the "Tasks" menu item.
+   *
+   * @return bool
+   */
+  public function upgrade_1033() {
+    $params = ['return' => 'id', 'name' => 'tasksassignments_administer'];
+    $parentId = (int) civicrm_api3('Navigation', 'getvalue', $params);
+    $params = ['return' => 'id', 'name' => 'ta_settings'];
+    $taSettingsId = (int) civicrm_api3('Navigation', 'getvalue', $params);
+    $permission = 'administer CiviCase';
+
+    // Update permissions
+    foreach ([$parentId, $taSettingsId] as $navId) {
+      $params = ['permission' => $permission, 'id' => $navId];
+      civicrm_api3('Navigation', 'create', $params);
+    }
+
+    // Update parent weight
+    civicrm_api3('Navigation', 'create', ['id' => $parentId, 'weight' => -97]);
```

Comment:
We can actually update permission and weight of the parentID in one call and the permission of taSettingsId in another call, which means we will call the API two times rather than three.
We can have a function do this that will optionally update the permission or not if present